Missionframework/scripts/shared/functions/F_kp_checkGear.sqf

```diff
@@ -19,44 +19,89 @@
 params [["_backpack", ""]];
 
 private _return = true;
-private _playerItems = [];
-if ((headgear player) != "") then {_playerItems pushBack (headgear player);};
-if ((goggles player) != "") then {_playerItems pushBack (goggles player);};
-if ((uniform player) != "") then {_playerItems pushBack (uniform player);};
-if ((vest player) != "") then {_playerItems pushBack (vest player);};
-if (((backpack player) != "") && ((backpack player) != _backpack)) then {_playerItems pushBack (backpack player);};
-{_playerItems pushBackUnique _x;} forEach (assignedItems player);
-{_playerItems pushBackUnique _x;} forEach (uniformItems player);
-{_playerItems pushBackUnique _x;} forEach (vestItems player);
-{_playerItems pushBackUnique _x;} forEach (backpackItems player);
-{_playerItems pushBackUnique _x;} forEach (weapons player);
-{if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (primaryWeaponItems player);
-{if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (secondaryWeaponItems player);
-{if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (handgunItems player);
+private _removedItems = [];
 
+if !(toLower (headgear player) in KP_liberation_allowed_items) then {
+    _removedItems pushBack (headgear player);
+    removeHeadgear player;
+};
+if !(toLower (goggles player) in KP_liberation_allowed_items) then {
+    _removedItems pushBack (goggles player);
+    removeGoggles player;
+};
+if !(toLower (uniform player) in KP_liberation_allowed_items) then {
+    _removedItems pushBack (uniform player);
+    removeUniform player;
+};
+if !(toLower (vest player) in KP_liberation_allowed_items) then {
+    _removedItems pushBack (vest player);
+    removeVest player;
+};
+if (!(toLower (backpack player) in KP_liberation_allowed_items) && ((backpack player) != _backpack)) then {
+    _removedItems pushBack (backpack player);
+    removeBackpack player;
+};
+
+private _playerItems = assignedItems player;
+_playerItems append ((getItemCargo (uniformContainer player)) select 0);
+_playerItems append ((getItemCargo (vestContainer player)) select 0);
```

Comment:
Wouldn't it be sensible to use `vestItems` here?
https://community.bistudio.com/wiki/vestItems

Reply:
The reason I chose to use `vestContainer`/`getItemCargo` combo was to match it with `removeItems`. `vestItems` would be simpler to read but it would also return all weapons and magazines which would then be given to `removeItems` which just doesn't do anything

Reply:
Yeah, it's not that good to read, but indeed the "more reliable way" you've chosen.